pydantic/fields.py

```diff
@@ -251,7 +257,7 @@ def _populate_sub_fields(self) -> None:  # noqa: C901 (ignore complexity)
         else:
             raise TypeError(f'Fields of type "{origin}" are not supported.')
 
-        if getattr(self.type_, '__origin__', None):
+        if has_subfields:
```

Comment:
looks like `has_subfields` will always be true here.

Reply:
Hmm yes, changed.

However I don't really understand the reason of the `if getattr(self.type_, '__origin__', None):` line that was there.. It seems logical that all container types have a notion of sub_fields so I think my change makes sense and it does not break any tests.  (Well except for one that basically seemed to be cause by another bug that was covered by this)

Reply:
@samuelcolvin  If you are OK with this change I consider this fixed..

Reply:
I think this is fine, the only tests which fails when the "if" is removed is fixed by the change to `flatten_errors`.

Reply:
I think this is fine, the only tests which fails when the "if" is removed is fixed by the change to `flatten_errors`.